core/src/player.rs

```diff
@@ -556,6 +556,99 @@ impl Player {
         self.is_playing
     }
 
+    fn get_allowed_context_menu_items<'gc>(
+        context: &mut UpdateContext<'_, 'gc, '_>,
+    ) -> Vec<&'static str> {
+        let mut activation = Activation::from_stub(
+            context.reborrow(),
+            ActivationIdentifier::root("[ContextMenu]"),
+        );
+
+        let dobj = activation.context.stage.root_clip();
+        let is_multiframe_movie = dobj
+            .as_movie_clip()
+            .map(|mc| mc.total_frames() > 1)
+            .unwrap_or(false);
```

Comment:
Looks like this should be based on the SWF header `swf.header.num_frames`, not on the actual root object. `stage.removeChild(this);`, for example, still shows the playhead menu items.

Reply:
Thinking about this some more, it depends on how much we want to match the Flash Player's behavior. If we do care about matching, then we'd check `header.swf.num_frames`. But if we don't really care, actually checking for a root MC >1 frames would be "better" (Play/Rewind/etc. would have no effect if it was shown in these cases, so why show them?). For the majority of content, it'll be the same either way.